src/WaveTrack.cpp

```diff
@@ -625,28 +790,18 @@ void WaveTrack::Trim (double t0, double t1)
 {
    bool inside0 = false;
    bool inside1 = false;
-   //Keeps track of the offset of the first clip greater than
-   // the left selection t0.
-   double firstGreaterOffset = -1;
 
    for (const auto &clip : mClips)
    {
-      //Find the first clip greater than the offset.
-      //If we end up clipping the entire track, this is useful.
-      if(firstGreaterOffset < 0 &&
-            clip->GetStartTime() >= t0)
-         firstGreaterOffset = clip->GetStartTime();
-
-      if(t1 > clip->GetStartTime() && t1 < clip->GetEndTime())
+      if(t1 > clip->GetPlayStartTime() && t1 < clip->GetPlayEndTime())
```

Comment:
The function `WaveTrack::Trim` has another interesting non-trivial change of behavior, like the test for correspondence, which is worthy of a comment.  Instead of deleting material, as before, now we hide the material in the smart clip but do not forget it.

I looked for where this function is used.  One place is in EffectSoundTouch -- but it always applies to a newly generated temporary track.  The other implements the Trim editing command, in EditMenus.cpp.  There is a comment there, no longer appropriate, which should be changed:

```
//Delete the section before the left selector
```